### PR TITLE
feat: Added MessageBird SMS client

### LINKED ISSUE #4386: [NV-2966] 🚀 Feature: Add Messagebird SMS Provider
### 🔖 Feature description

Use the [messagebird node sdk](https://github.com/messagebird/messagebird-nodejs) and the [messagebird docs](https://developers.messagebird.com/api/sms-messaging/#send-outbound-sms) to create a new provider while following [Novu's guide to a new provider](https://docs.novu.co/community/add-a-new-provider#how-to-add-a-new-provider)

### 🎤 Why is this feature needed ?

have more providers for the community

### ✌️ How do you aim to achieve this?

community contributions

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-2966](https://linear.app/novu/issue/NV-2966/🚀-feature-add-messagebird-sms-provider)</sub>

### PR DESCRIPTION
**Pull request by ayushshukla7777**
### What change does this PR introduce?
This commit adds the MessageBird SMS client to our project.

### Why was this change needed?

Closes #4386

### Other information (Screenshots)
image
image
image

### CODE REVIEW

File: packages/application-generic/package.json [hunk 1]
```diff
@@ -98,6 +98,7 @@
     "@novu/telnyx": "^0.21.0",
     "@novu/termii": "^0.21.0",
     "@novu/testing": "^0.21.0",
+    "@novu/messagebird": "^0.16.3",
```
Comment (Cliftonz) [inline packages/application-generic/package.json:101]: version should be the same as the rest

File: providers/messagebird/package.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
+{
+  "name": "@novu/messagebird",
+  "version": "0.16.3",
```
Comment (Cliftonz) [inline providers/messagebird/package.json:3]: Same as above

File: providers/messagebird/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
+# Novu Messagebird Provider
+
+A Messagebird sms provider library for [@novu/node](https://github.com/novuhq/novu)
+
+## Usage
+
+# MessageBird Node.js API Client
```
Comment (Cliftonz) [inline providers/messagebird/README.md:7]: It looks like you just copy pasted the docs for the client, please look at some of the other providers and make this Novu centric.

**Review by Cliftonz — CHANGES_REQUESTED** [review]

**Comment by ayushshukla7777** [comment]
hey @Cliftonz , Did the changes as you requested.

**Comment by Cliftonz** [comment]
Looks like the cspell pipeline failed, you will need to add these two words will need to be added to the cspell list.
CleanShot 2023-10-28 at 16 54 19@2x

**Comment by ayushshukla7777** [comment]
hey @Cliftonz , Added them to cspell

**Review by Cliftonz — APPROVED** [review]

**Pull request merged**